### DISCUSSION TITLE
`x-intersect` doesn't catch intersection on slow scrolling

### DISCUSSION THREAD

**Discussion by dmitry-ivanov** [alpinejs/alpine#1833] [7. Bugs (archived)] [answered]
Hi,

Thank you for your hard work on Alpine 3 and all the good new stuff! 👍

I've noticed that `x-intersect` doesn't catch intersection if you scroll really slowly _(see the gray box on the `gif` below)_:

```html
<div class="border border-yellow-400 h-40 m-4"></div>

<div x-data="{ shown: false }" x-intersect="shown = true; console.log('Caught intersection! 👍')" class="border p-4 m-4">
    <div x-show="shown" x-transition.duration.1000ms>
        I'm in the viewport!
    </div>
</div>
```

![x-intersect](https://user-images.githubusercontent.com/1286821/126712086-ffb74e3c-5402-49d0-8475-8fd0f7dc4a4e.gif)

I understand that it has to do with the `IntersectionObserver API` _(which is used under the hood)_, but **maybe we could fine-tune some options of it?** Because, currently, if you're using `x-intersect` for transitions - there's a chance that those transitions would not be started, thus making the UI broken.

Would appreciate it if you have any ideas!

Thank you,
Have a great day!

**Comment by markjaquith** [comment]
I'm working a PR related to `x-intersect` and encountered this same behavior. The intersection logic seems faulty, because it relies on `intersectionRatio`, which is not reliable. You can have an `intersectionRatio` of `0` but be inside the viewport. You can have an intersection ratio of less than 100, but have the item covering the entirety of the viewport.

I'm working on figuring this out, but to anyone encountering this: yeah, it's kinda flaky right now.

**Comment by SimoTod** [comment]
X-intersect:enter has a different implementation that should catch slow scrolling.

**Reply by markjaquith** [reply]
Yep. It does. There's no reason for there to be a working implementation and a broken implementation, so this PR properly unifies them: #2557

**Comment by markjaquith** [comment, marked as answer]
#2557 was merged and fixes this issue.
